kmip/tests/unit/pie/test_client.py

```diff
@@ -0,0 +1,598 @@
+# Copyright (c) 2015 The Johns Hopkins University/Applied Physics Laboratory
+# All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+import mock
+import six
+import testtools
+
+from kmip.core import attributes as attr
+from kmip.core import enums
+from kmip.core import objects as obj
+
+from kmip.core.factories import attributes
+from kmip.core.messages import contents
+
+from kmip.services.kmip_client import KMIPProxy
+from kmip.services import results
+
+from kmip.pie.client import KmipClient
+
+from kmip.pie.exceptions import ClientConnectionFailure
+from kmip.pie.exceptions import ClientConnectionNotOpen
+from kmip.pie.exceptions import KmipOperationFailure
+
+from kmip.pie import factory
+from kmip.pie import objects
+
+
+class TestKmipClient(testtools.TestCase):
+    """
+    Test suite for the KmipClient.
+    """
+
+    def setUp(self):
+        super(TestKmipClient, self).setUp()
+        self.attribute_factory = attributes.AttributeFactory()
+
+    def tearDown(self):
+        super(TestKmipClient, self).tearDown()
+
+    @mock.patch('kmip.pie.client.KMIPProxy',
+                mock.MagicMock(spec_set=KMIPProxy))
+    def test_init(self):
+        """
+        Test that a KMIPClient can be constructed with valid arguments.
+        """
+        KmipClient(
+            hostname='127.0.0.1',
+            port=5696,
+            cert='/example/path/to/cert',
```

Comment:
Seeing this here I'm wondering if we should add checks in constructor to validate that files exist before returning a new instance. This would alert up front about issue with files. Otherwise I imagine the caller will receive socket exception or something.